integration/query_array_test.go

```diff
@@ -212,6 +212,90 @@ func TestQueryArrayDotNotation(t *testing.T) {
 
 			cursor, err := collection.Find(ctx, tc.filter, options.Find().SetSort(bson.D{{"_id", 1}}))
 			if tc.err != nil {
+				require.Nil(t, tc.expectedIDs)
+				AssertEqualError(t, *tc.err, err)
+				return
+			}
+			require.NoError(t, err)
+
+			var actual []bson.D
+			err = cursor.All(ctx, &actual)
+			require.NoError(t, err)
+			assert.Equal(t, tc.expectedIDs, CollectIDs(t, actual))
+		})
+	}
+}
+
+func TestQueryElemMatchOperator(t *testing.T) {
+	t.Parallel()
+	ctx, collection := setup(t, shareddata.Scalars, shareddata.Composites)
+
+	for name, tc := range map[string]struct {
```

Comment:
All the test cases have `gt` or `lt`. Can we have one that is a mix of conditions, something like `{ value: { $elemMatch: { field1: "a", field2: { $gt: 42 } } } }`?

Also, it seems that `$ne` might give interesting cases, and we should add it to the tests.

Reply:
I've added a similar case to that one you suggest → `GtField`.
Also please see https://github.com/FerretDB/FerretDB/pull/707#issuecomment-1152037874